### PR TITLE
gdk-pixbuf: rename from gdk_pixbuf

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
@jtojnar mentioned doing this in https://github.com/NixOS/nixpkgs/issues/56966#issuecomment-470147795
so I guess that time is now.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jtojnar** [comment]
Cannot review at the moment but the main problem I had with this the last time I attempted this is that `gdk_pixbuf` appears in patches as a part of projects’ source code very often. And also since `gdk-pixbuf` is not a valid bash variable, we cannot use it in `substituteAll`.

File: pkgs/applications/window-managers/way-cooler/crates-io.nix [hunk 1]
```diff
@@ -546,76 +546,76 @@ rec {
 # end
 # gdk-pixbuf-0.2.0
 
-  crates.gdk_pixbuf."0.2.0" = deps: { features?(features_.gdk_pixbuf."0.2.0" deps {}) }: buildRustCrate {
+  crates.gdk-pixbuf."0.2.0" = deps: { features?(features_.gdk-pixbuf."0.2.0" deps {}) }: buildRustCrate {
```
Comment (worldofpeace) [inline pkgs/applications/window-managers/way-cooler/crates-io.nix]: **Note**: I probably need to regenerate this file.

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 2]
```diff
@@ -92057,7 +92057,7 @@ self: {
      }) {inherit (pkgs) gtk3;};
 
   "gi-gdkpixbuf" = callPackage
-    ({ mkDerivation, base, bytestring, Cabal, containers, gdk_pixbuf
+    ({ mkDerivation, base, bytestring, Cabal, containers, gdk-pixbuf
```
Comment (Infinisil) [inline pkgs/development/haskell-modules/hackage-packages.nix]: You'll have to change this in `hackage2nix` here: https://github.com/NixOS/cabal2nix/blob/f4355aa45723abd15b29eb826800839278f91ed9/src/Distribution/Nixpkgs/Haskell/FromCabal/Name.hs#L50

Fixing it in this file will just make it reappear the next time it's regenerated.

Comment (worldofpeace) [reply pkgs/development/haskell-modules/hackage-packages.nix]: Opened https://github.com/NixOS/cabal2nix/pull/424

**Comment by zimbatm** [comment]
It would be best to start by making an alias. And keep it like that for one release. That way it will be easier to back-port security fixes.

**Comment by worldofpeace** [comment]
@zimbatm This pr has always included an alias in `aliases.nix`.

**Comment by zimbatm** [comment]
I mean one solution would be to merge the alias now, and then do the rename and swap the aliases in the next release.

**Comment by worldofpeace** [comment]
@zimbatm If I were to merge that alias in `aliases.nix` wouldn't that cause `eval-package-list-no-aliases` to fail indefinitely until all references to `gdk_pixbuf` are removed from nixpkgs?

**Comment by zimbatm** [comment]
@worldofpeace I don't know. Sorry for holding the PR back, I guess renaming things directly will be ok too.

**Comment by worldofpeace** [comment]
@peti would it be appropriate after this is merged to manually update the haskell package set?
I think that's the way to go about this. You'd need to merge https://github.com/NixOS/cabal2nix/pull/424

**Comment by peti** [comment]
> @peti would it be appropriate after this is merged to manually update the haskell package set?

Yes, I think so.

>You'd need to merge [NixOS/cabal2nix#424](https://github.com/NixOS/cabal2nix/pull/424)

Done.

**Comment by worldofpeace** [comment]
@peti Is merged :+1:

**Comment by FRidh** [comment]
evaluation of staging is failing:
https://gist.github.com/GrahamcOfBorg/d4200600779484deceb3c72071a9fe87

**Comment by worldofpeace** [comment]
Yeah that's expected, it needs to be manually updated...

**Comment by FRidh** [comment]
What do you mean expected? Failing builds is one thing, but failing evaluation is another. Can we do a simple sed substitute?

**Comment by worldofpeace** [comment]
You can't/shouldn't manually change anything in `haskell-packages.nix`.
https://github.com/NixOS/cabal2nix/pull/424 was merged so it can be regenerated using the new name, but this change had to happen before that. If I were to just manually change it without that it would have regenerated using the old name.

Since I have no control over the haskell infustructure in nixpkgs and know very little of it, I'm expecting @peti to correct this and why I expected this to happen. I'm waiting for it to be corrected because it's out of my control.

**Comment by FRidh** [comment]
For now we can just do a substitute until. It does no harm and when the next time someone (likely @peti) generates the set it will be fine anyway.

**Comment by peti** [comment]
I won't re-generate the package set for at least one week or longer as I'm on vacation. Generally speaking, I never re-generate anything in `staging` where this PR went.

**Comment by jtojnar** [comment]
I would suggest manually `sed`ing `hackage-packages.nix` until then.

**Comment by worldofpeace** [comment]
#65487

**Pull request merged**